server/src/main/java/io/druid/query/lookup/MapLookupExtractorFactory.java

```diff
@@ -101,4 +117,29 @@ public int hashCode()
     result = 31 * result + (isOneToOne ? 1 : 0);
     return result;
   }
+
+  public static class MapLookupIntrospection implements LookupIntrospectHandler
+  {
+    final private Map<String, String> map;
```

Comment:
Suggest making a weak reference map so that it can be garbage collected if the `MapLookupExtractorFactory` gets replaced